crypto/keys/ecdsa/privkey.go

```diff
@@ -0,0 +1,128 @@
+package ecdsa
+
+import (
+	"crypto/ecdsa"
+	"crypto/rand"
+	"crypto/sha256"
+	"fmt"
+	"math/big"
+
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	gogotypes "github.com/gogo/protobuf/types"
+)
+
+type ecdsaSK struct {
+	ecdsa.PrivateKey
+}
+
+// GenSecp256r1 generates a new secp256r1 private key. It uses operating system randomness.
+func GenSecp256r1() (cryptotypes.PrivKey, error) {
+	key, err := ecdsa.GenerateKey(secp256r1, rand.Reader)
+	return &ecdsaSK{*key}, err
```

Comment:
```suggestion
	if err != nil {
		return nil, err
	}

	return &ecdsaSK{*key}, nil
```

Reply:
if you have an error, then the value other value is not valid by definition. So how this change improves the behavior?

Reply:
https://lgtm.com/projects/g/cosmos/cosmos-sdk/rev/pr-85c650b62f3222e95ab7a1af3a865fcdb357838d

Reply:
`key` may be nil